acme/authority.go

```diff
@@ -54,8 +57,24 @@ var (
 	orderTable             = []byte("acme_orders")
 	ordersByAccountIDTable = []byte("acme_account_orders_index")
 	certTable              = []byte("acme_certs")
+	ordinal                int
 )
 
+// Ordinal is used during challenge retries to indicate ownership.
+func init() {
+	ordstr := os.Getenv("STEP_CA_ORDINAL")
+	if ordstr == "" {
+		ordinal = 0
+	} else {
+		ord, err := strconv.Atoi(ordstr)
+		if err != nil {
+			log.Fatal("Unrecognized ordinal ingeter value.")
+			panic(nil)
```

Comment:
why are we panicking? This method should return an error and be handled just like any other error.

If we'd like to switch to using more panic's we can have that discussion. That would be a pretty significant change and I think we'd rather handle errors in a uniform way. If anything, that can be part of another PR.

Reply:
There's nothing to return and the entire program can't continue execution if this value can't be parsed. 

In general, I'm we should fail loudly and obnoxiously for anything that isn't spurious. For programmer or operator logic/configuration/usage errors, we should know immediately. For something like a read error that can happen during normal execution of the code, then yes internal server errors displayed in the UI (the request response) are appropriate.

Reply:
Yep, agree that there's nothing else you could be returning. This was more in reference to `panic()` being used in other places.

That said, https://github.com/smallstep/certificates/blob/master/cmd/step-ca/main.go#L143-L149 is a good example of how we should exit. Choose where the output is going and what status code to show the user.

Reply:
This is an init, so it's ok to panic, in fact, I believe log.Fatal will panic, so it will never reach the panic, we probably want to be more explicit in the error, show, what the value was, and probably the environment variable.

Reply:
So you'd prefer `log.Fatal` to `panic` here?

Reply:
I mean, it will never reach panic there, https://golang.org/src/log/log.go?s=6917:6957#L190

Reply:
I meant "replace panic with log.fatal".

Reply:
Also, this would be better in `NewAuthority` so I'll move it there.